### PR TITLE
Migrate ResetPassword form to formik

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/26178

As with other formik migration PRs, one of the added benefits is that it's now possible to fully unit test the form without complex mocking.

How to test:
- Setup email in Admin -> Settings
- Logout, then on the login screen click forgot your password, enter your email, and submit the form
- Click on the password reset link in the email
- You should be navigated to the ResetPassword form
- Make sure it shows error message for common passwords, like `admin123`
- Make sure it is possible to reset the password with the form

Screenshot 2022-11-02 at 16 15 22

### CODE REVIEW

File: frontend/src/metabase/auth/components/ResetPasswordForm/ResetPasswordForm.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+const ResetPasswordForm = ({
+  onValidatePassword,
+  onSubmit,
+}: ResetPasswordFormProps): JSX.Element => {
+  const initialValues = useMemo(
+    () => ({ password: "", password_confirm: "" }),
+    [],
+  );
```
Comment (iethree) [inline frontend/src/metabase/auth/components/ResetPasswordForm/ResetPasswordForm.tsx:41]: I don't think we need useMemo here? and I think it's less performant for static values like this.

File: frontend/src/metabase/auth/components/ResetPasswordForm/ResetPasswordForm.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+const ResetPasswordSchema = Yup.object({
+  password: Yup.string()
+    .required(t`required`)
+    .test(async (value = "", context) => {
+      const error = await context.options.context?.onValidatePassword(value);
+      return error ? context.createError({ message: error }) : true;
+    }),
+  password_confirm: Yup.string()
+    .required(t`required`)
+    .oneOf([Yup.ref("password")], t`passwords do not match`),
```
Comment (iethree) [inline frontend/src/metabase/auth/components/ResetPasswordForm/ResetPasswordForm.tsx:26]: I really like this `Yup` business!

**Review by iethree — APPROVED** [review]
💖

File: frontend/src/metabase/auth/components/ResetPasswordForm/ResetPasswordForm.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/auth/components/ResetPasswordForm/ResetPasswordForm.tsx:41]: We don't need it here. However we *might* need it case we use `enableReinitialize` in formik. I just thought it's a good idea to make sure that `initialValues`'s reference is always stable.

**Pull request merged**